experiment/compatibility-versions/kind-upgrade.sh

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bash
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# script adapted from - https://gist.github.com/aojea/2c94034f8e86d08842e5916231eb3fe1
+
+set -e
+set -o pipefail
+
+build_docker(){
+  build/run.sh make all WHAT="cmd/kubectl cmd/kubelet" 1> /dev/null
+  make quick-release-images 1> /dev/null
+}
+
+update_kubelet() {
+ for n in $NODES; do
+   # Backup previous kubelet
+   docker exec $n cp /usr/bin/kubelet /usr/bin/kubelet.bak
```

Comment:
Todo: use something like `which kubelet` to avoid depending on the exact path.

https://kind.sigs.k8s.io/docs/design/node-image/